test/position-managers/PositionManager.t.sol

```diff
@@ -0,0 +1,263 @@
+
+        lpm = new PositionManager(manager);
+
+        IERC20(Currency.unwrap(currency0)).approve(address(lpm), type(uint256).max);
+        IERC20(Currency.unwrap(currency1)).approve(address(lpm), type(uint256).max);
+    }
+
+    function test_modifyLiquidities_reverts_mismatchedLengths() public {
+        Planner.Plan memory planner = Planner.init();
+        planner = planner.add(Actions.MINT, abi.encode("test"));
+        planner = planner.add(Actions.BURN, abi.encode("test"));
+
+        bytes[] memory badParams = new bytes[](1);
+
+        vm.expectRevert(IPositionManager.MismatchedLengths.selector);
+        lpm.modifyLiquidities(abi.encode(planner.actions, badParams), block.timestamp + 1);
+    }
+
+    function test_mint_withLiquidityDelta(IPoolManager.ModifyLiquidityParams memory params) public {
+        params = createFuzzyLiquidityParams(key, params, SQRT_PRICE_1_1);
+        // liquidity is a uint
+        uint256 liquidityToAdd =
+            params.liquidityDelta < 0 ? uint256(-params.liquidityDelta) : uint256(params.liquidityDelta);
+        LiquidityRange memory range =
+            LiquidityRange({poolKey: key, tickLower: params.tickLower, tickUpper: params.tickUpper});
+
+        uint256 balance0Before = currency0.balanceOfSelf();
+        uint256 balance1Before = currency1.balanceOfSelf();
+
+        uint256 tokenId = lpm.nextTokenId();
+        BalanceDelta delta = mint(range, liquidityToAdd, address(this), ZERO_BYTES);
+
+        assertEq(tokenId, 1);
+        assertEq(lpm.ownerOf(tokenId), address(this));
+
+        bytes32 positionId =
+            keccak256(abi.encodePacked(address(lpm), range.tickLower, range.tickUpper, bytes32(tokenId)));
+        (uint256 liquidity,,) = manager.getPositionInfo(range.poolKey.toId(), positionId);
+
+        assertEq(liquidity, uint256(params.liquidityDelta));
+        assertEq(balance0Before - currency0.balanceOfSelf(), uint256(int256(-delta.amount0())), "incorrect amount0");
+        assertEq(balance1Before - currency1.balanceOfSelf(), uint256(int256(-delta.amount1())), "incorrect amount1");
+    }
+
+    function test_mint_exactTokenRatios() public {
+        int24 tickLower = -int24(key.tickSpacing);
+        int24 tickUpper = int24(key.tickSpacing);
+        uint256 amount0Desired = 100e18;
+        uint256 amount1Desired = 100e18;
+        uint256 liquidityToAdd = LiquidityAmounts.getLiquidityForAmounts(
+            SQRT_PRICE_1_1,
+            TickMath.getSqrtPriceAtTick(tickLower),
+            TickMath.getSqrtPriceAtTick(tickUpper),
+            amount0Desired,
+            amount1Desired
+        );
+
+        LiquidityRange memory range = LiquidityRange({poolKey: key, tickLower: tickLower, tickUpper: tickUpper});
+
+        uint256 balance0Before = currency0.balanceOfSelf();
+        uint256 balance1Before = currency1.balanceOfSelf();
+
+        uint256 tokenId = lpm.nextTokenId();
```

Comment:
does feel like `mint` should return the `tokenId` that was minted. If you imagine 10 txs go in quick succession minting, how are you meant to find your ID easily

Reply:
so we did this because we wanted the actions inside the unlockCallback to return all the same data... also if we combine mint with increase then increase will always have to return the tokenId which is suboptimal imo

Reply:
leaving for now.. can continue in our return variables discussion